sdcm/logcollector.py

```diff
@@ -220,7 +220,12 @@ class MonitoringStack(BaseLogEntity):
 
     @staticmethod
     def get_monitoring_base_dir(node):
-        return node.parent_cluster.monitor_install_path_base
+        # Avoid cyclic dependencies
+        from sdcm.cluster import BaseMonitorSet
+        if hasattr(node, "parent_cluster") and node.parent_cluster:
```

Comment:
Another thing, why do you check if a node has a `parent_cluster` attribute? All node classes inherit from BaseNode and  `parent_cluster` attribute is defined on the BaseNode, so how can it happen that a node doesn't have a  `parent_cluster` attribute?

Reply:
yes, this situation is usual, if run hydra collect-log. in this case we collect only nodes and don't use cluster instance